### DISCUSSION TITLE
Save the Upload File URL

### DISCUSSION THREAD

**Discussion by moskoweb** [filamentphp/filament#5951] [Help]
How can I save the URL of the file that was uploaded with FileUpload to the database?

Is there any way to do this directly through the element, or just with treatment by pressing "save"?

**Comment by danharrin** [comment]
It should work out of the box, the file name is saved.

**Comment by moskoweb** [comment]
But wouldn't it be possible to do that after finishing the upload, the value of the field is the URL of the file, instead of the name? Or maybe the file path.

**Reply by danharrin** [reply]
why would you store the URL? that makes it fragile, what happens if you change domain or move directory? use a getter method to generate the URL on demand

**Comment by moskoweb** [comment]
We store it directly in storage, so the URL is always fixed, and we save
the URL because it's used in static elements.

And also with customer company cdn.

**Reply by danharrin** [reply]
I don't understand what you're asking me to do

**Reply by moskoweb** [reply]
I saw that it has a getUploadedFileUrlUsing function, which I thought would be able to use it to set the value of a field with the value of the URL of the file.

However, I tested some forms of function, and it does not return anything in the method.

```php
Components\FileUpload::make('file')
    ->getUploadedFileUrlUsing(fn ($file, callable $set) => $set('file_url', $file))
```

**Comment by ryo-kozin** [comment]
I'd like to know how to save the updated file path to the column.

**Reply by ryo-kozin** [reply]
I just forgot to set the column name to fillable attribute on the model.
